src/diric.jl

```diff
@@ -0,0 +1,55 @@
+export diric
+
+"""
+    kernel = diric(Ω::Real, n::Int; atol)
+
+Dirichlet kernel, also known as periodic sinc function,
+where `n` should be a positive integer.
+Returns `sin(Ω * n/2) / (n * sin(Ω / 2))` typically,
+but `±1` when `Ω` is a multiple of 2π.
+
+In the usual case where 'n' is odd, the output is equivalent to
+``1/n \\sum_{k=-(n+1)/2}^{(n+1)/2} e^{i k Ω}``,
+which is the discrete-time Fourier transform (DTFT)
+of a `n`-point moving average filter.
+
+When `n` is odd or even, the function is 2π or 4π periodic, respectively.
+
+When `Ω` is an `AbstractFloat` (e.g., `Float32` or `BigFloat`),
+the return type matches that of `Ω`.  Otherwise the return type is `Float64`.
+
+As of 2021-03-19, the Wikipedia definition has different factors.
+The definition here is consistent with scipy and other software frameworks.
+
+This implementation treats inputs near multiples of 2π fairly carefully.
+If the denominator `sin(Ω/2)` is close to 0, then this function uses
+an accurate 2nd-order Taylor expansion rather than simple division. 
+
+The optional argument `atol` defaults to `sqrt(eps(T))`
+and is used to assess whether `sin(Ω/2) ≈ 0`.
+Here `T` is the type of the floating-point argument `Ω`.
+This default tolerance originates from [`rtoldefault`](@ref)
+used in [`isapprox`](@ref).
```

Comment:
Do these references actually work in some context? Building the docs locally, I get warnings about them during build and dead links in the output.

Reply:
Oops, I was following some examples in the base julia docs without fully understanding them and now I realize that probably such refs only work within one package, not between packages.  I will remove them.